worksheet.rb

```diff
@@ -23,6 +24,12 @@
 # into this data structure, such as "DR0004"
 # and "3rd Feb 2016" and "RD0022"
 
+rides = [{driver_id:"DR0001", date:["3rd Feb 2016", "3rd Feb 2016", "5th Feb 2016"], cost:[10, 30, 45], ride_id:["RD0003","RD0015","RD0003"], rating:[3, 4, 2]},
```

Comment:
👍 , this works, but you have basically 3 parallel arrays in the dates, costs, ride_id and ratings.  You have to keep the indexes aligned.  

This works for the given examples, but for more general-purpose problems this might work as well.   Each entry is labeled and relatively easy identify what value goes with each item. 

```suggestion
driver_data = {
  dr0001: [
    {
      date: "3rd Feb 2016",
      cost: [10],
      ride_id: "RD0003",
      rating: 3,
    }
    ...
  ],
dr0002: ...
 
```